src/pkg/cli/cert.go

```diff
@@ -140,33 +171,62 @@ func waitForCNAME(ctx context.Context, domain, albDns string) error {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-ticker.C:
-			cname, err := waitForCNAMEInSync(ctx, domain)
-			cname = strings.TrimSuffix(cname, ".")
-			if err != nil || strings.ToLower(cname) != strings.ToLower(albDns) {
-				if !msgShown {
-					term.Infof(" * Please setup CNAME record for %v", domain)
-					fmt.Printf("  %v  CNAME  %v\n", domain, strings.ToLower(albDns))
-					term.Infof(" * Waiting for CNAME record setup and DNS propagation...")
-					msgShown = true
-				}
-				if doSpinner {
-					fmt.Print(spin.Next())
-				}
-			} else {
+			if checkDomainDNSReady(ctx, domain, albDns) {
 				return nil
 			}
+			if !msgShown {
+				term.Infof(" * Please setup CNAME record for %v", domain)
+				fmt.Printf("  %v  CNAME  %v\n", domain, strings.ToLower(albDns))
+				term.Infof(" * Waiting for CNAME record setup and DNS propagation...")
+				msgShown = true
+			}
+			if doSpinner {
+				fmt.Print(spin.Next())
+			}
 		}
 	}
 }
 
+// The DNS is considered ready if the CNAME of the domain is pointing to the ALB domain and in sync
+// OR if the A record of the domain is pointing to the same IP addresses of the ALB domain and in sync
+func checkDomainDNSReady(ctx context.Context, domain, expectedCNAME string) bool {
+	expectedCNAME = strings.TrimSuffix(expectedCNAME, ".")
+	cname, err := getCNAMEInSync(ctx, domain)
+	term.Debugf(" - CNAME for %v is %v: %v", domain, cname, err)
+	// Ignore other types of DNS errors
+	if err == DNSNotInSyncError {
+		return false
+	}
+	cname = strings.TrimSuffix(cname, ".")
+	if strings.EqualFold(cname, expectedCNAME) {
```

Comment:
allow more than 1 CNAME. In particular, allow public FQDN with and without --port